src/Socket.cpp

```diff
@@ -240,6 +255,31 @@ MessagePtr Socket::takeNextMessage()
     }
 }
 
+MessagePtr Socket::takeNextMessageBlocking()
```

Comment:
It looks like `takeNextMessage` is now unused. We should remove the old implementation. Better still, rename `takeNextMessageBlocking` to `takeNextMessage` as well, so that all of these calls are blocking. They are expected to be blocking by both CuraEngine and ScannerEngine.